### PR TITLE
additional check for safer_getattr 

### PR DESCRIPTION
**Pull request by loechel**
add an additional check for potential breakout capability via Inspection Attributes Names in the provided safer_getattr method.

### CODE REVIEW

File: CHANGES.rst [hunk 1]
```diff
@@ -6,6 +6,8 @@ Changes
 
 - Allow to use the package with Python 3.13 -- Caution: No security
   audit has been done so far.
+- Add an additional check for potential breakout capability via inspection attribute names
+  in the provided ``safer_getattr`` method that is part of the ``safer_builtins``.
```
Comment (d-maurer) [inline CHANGES.rst:10]: There is no `safer_builtins`.
```suggestion
- Increase the safety level of ``safer_getattr``  allowing applications to use it as ``getattr`` implementation. Such use should now follow the same policy and give the same level of protection as direct attribute access in an environment based on ``RestrictedPython``'s ``safe_builtints``. 
```

**Review by d-maurer — APPROVED** [review]

**Review by icemac — CHANGES_REQUESTED** [review]
The mixed in reformatting of the code makes it unneccessarily hard to review this PR. Please reduce it to the actual changes needed.

**Comment by loechel** [comment]
@icemac I have undone the linting changes, please review again.

File: src/RestrictedPython/Guards.py [hunk 2]
```diff
@@ -253,6 +254,11 @@ def safer_getattr(object, name, default=None, getattr=getattr):
             (isinstance(object, type) and issubclass(object, str))):
         raise NotImplementedError(
             'Using the format*() methods of `str` is not safe')
+    if name in INSPECT_ATTRIBUTES:
+        raise AttributeError(
+            f'"{name}" is a restricted name,'
+            ' that is forbidden to access in RestrictedPython.',
+        )
```
Comment (icemac) [inline src/RestrictedPython/Guards.py:261]:
```suggestion
            ' that is forbidden to access in RestrictedPython.')
```

**Review by icemac — APPROVED** [review]

**Pull request merged**